tcpdirect.cu

```diff
@@ -0,0 +1,785 @@
+#ifdef WITH_TCPDIRECT
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include <asm-generic/errno-base.h>
+#include <asm-generic/socket.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <linux/if.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <linux/dma-buf.h>
+
+#include <memory>
```

Comment:
These are C++ headers. I would recommend sticking to C. Is there a reason C++ is involved? I also notice you moved the compiler to g++.

I would recommend sticking to C just becasue the rest of neper is C. A bit weird to add C++ code here.